### PR TITLE
Add lexer and style for LilyPond

### PR DESCRIPTION
**Pull request by Jean-Abou-Samra**
GNU LilyPond is a text-based music typesetter. Because its concepts
are completely different from programming languages, this adds
a special-purpose to highlight the special tokens.

The SchemeLexer is expanded to gain the ability of detecting when
one expression stops. LilyPondLexer subclasses SchemeLexer.

Builtins (the most important part) are generated with a script
put in external/ (as was already done for Lasso).

As part of this change, the CPSALexer is made no longer to
inherit from SchemeLexer. The inheritance was unused anyway.

### CODE REVIEW

File: pygments/lexers/_lilypond_builtins.py [hunk 1]
```diff
@@ -0,0 +1,5148 @@
...
+  "a-natural",
+  "a-sharp",
+  "a-sharpsharp",
+  "ab",
+  "acousticbassdrum",
+  "acousticsnare",
+  "ad",
+  "adb",
+  "add",
+  "aeh",
```
Comment (lemzwerg) [inline pygments/lexers/_lilypond_builtins.py:134]: Is it intentional that there are so many duplicates?

File: pygments/lexers/lilypond.py [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+    Token.Name.Builtin.MarkupCommand,
+]
+
+class LilyPondLexer(SchemeLexer):
+    """
+    Lexer for input to `LilyPond <https://lilypond.org>`_, a text-based music typesetter.
+
+    .. important::
+
+       This lexer is meant to be used in conjuction with the ``lilypond`` style.
```
Comment (lemzwerg) [inline pygments/lexers/lilypond.py:70]: `s/conjuction/conjunction/`

File: pygments/lexers/_lilypond_builtins.py (hunk 1 above)
Comment (Jean-Abou-Samra) [reply pygments/lexers/_lilypond_builtins.py:128]: They come from different languages. I've used `uniq-list` to avoid this.

**Comment by Jean-Abou-Samra** [comment]
This is now ready for review.

**Comment by Jean-Abou-Samra** [comment]
Is there anything I can do to help this PR move forward?

**Comment by Anteru** [comment]
No, I just need to find some time to review it :/

**Comment by Jean-Abou-Samra** [comment]
@anteru Gentle ping?

**Comment by Anteru** [comment]
It's on my radar, just busy with other stuff right now -- sorry this is taking that long :(

**Comment by Anteru** [comment]
Finally got around to this. Merged, thanks!

**Comment by Jean-Abou-Samra** [comment]
Thank you!!

**Comment by Jean-Abou-Samra** [comment]
@Anteru Unfortunately it seems that this PR has broken some checks that have been added in the meantime. I'm on it and will submit a PR very soon.

**Comment by Anteru** [comment]
I think 2d831c1e323696285bab470652c64b60874c00d7 fixes the last issues.

**Comment by Jean-Abou-Samra** [comment]
Yeah, you were faster than me :-) Thanks!

**Comment by Jean-Abou-Samra** [comment]
Ok, I still see two quirks:

- my Sphinx complains about the "LilyPond" target being duplicated in languages.rst (easy to fix),
- I see that a style gallery was added in the meantime. The LilyPond-specific style is not relevant there and doesn't look very attractive :-)  I'm not sure what's the best fix: should I just add a check for `style != 'lilypond'` in conf.py? Should I introduce a style_gallery_exclude attribute?

**Comment by Anteru** [comment]
Oh right, the style gallery -- good catch. I think an "exclude" attribute would be indeed the best solution. Could you please prepare a PR for both issues?

**Comment by Jean-Abou-Samra** [comment]
Will do.

**Pull request merged**